pp.c

```diff
@@ -8060,22 +8063,48 @@ PP(pp_multiparam)
                 hv_store_ent(hv, name, newSVsv(val), 0);
             }
             else {
-                // TODO: Consider collecting up all the names of unrecognised
-                // in one string
-                croak_caller("Unrecognized named parameter '%" UTF8f "' to subroutine '%" SVf "'",
-                    UTF8fARG(true, namelen, namepv), S_find_runcv_name());
+                n_errors++;
+                if(!error_names)
+                    error_names = sv_2mortal(newSVpvs(""));
+                if(n_errors <= 5) {
+                    /* Only bother collecting up the first 5 */
+                    if(n_errors > 1)
+                        sv_catpvs(error_names, ", ");
+                    sv_catpvf(error_names, "'%" UTF8f "'", UTF8fARG(true, namelen, namepv));
+                }
```

Comment:
I'd be inlined to put this duplicated logic in a function.
```
accumulate_error_names(&n_errors, &error_names, namepv, namelen);
```
but you might not be.

Reply:
Yeah, seems reasonable. Doesn't quite cut down the overall line count but it does remove the otherwise pointless and subtle near-repetition.